### DISCUSSION TITLE
"#OpenSourceJourneys" Challenge

### DISCUSSION THREAD

**Discussion by suitupstan** [EddieHubCommunity/support#3673] [Ideas] [answered]
# 💡 Introduction

A 100 days long, journey to get you into open-source 🎉

Every week you are given a set of challenges, about 7, you have to complete at least 1 per week.

Don't worry the challenges won't be any hard, they would be "Exploring GitHub Features"/"Making First Contributions" 🙂 

And there will be a monthly goal, on which the challenges will be based.

I picture it this way:
- First Month, Learn Git, GitHub & Open Source
- Second Month, Build a simple project, on a theme, like a hackathon, but with a whole month, you have a lot to explore!
- Third Month, Writing a blog post about your learnings & the project, Open Sourcify your repo and Contribution Raids

As you must have seen we are making it really slow-paced, to make sure every gets a lot of time for work/life balance, in other places where I have seen similar stuff they usually make it 7 days and then rush things, a workshop every few hours, projects should be built in 2 days, etc...

# ➡️ What would be EddieHub's contribution 👀 

Telling people what to do, isn't just it, EddieHub is one of the biggest open source communities, when EddieHub does it, it has to be special.

- We are going to be helping people the same way through our #help channel
- Our projects make up for great challenges because they are beginner-friendly!
- We will be having a leaderboard 🎉 
- Because EddieHub Ambassadors are rolling out soon, it is gonna make a huge difference, after the ambassadors are also involved!
- We will be featuring the best creations/achievements of the month in the EddieHub newsletter 🥳 
- The site for this will be open source, meaning the contributors choose the challenges 👀 
- Collaboration 1000000000000000000000000000000000000000000000000000000000% 💪 

# 🎁 Ending note

EddieHub got me into open source, and I want to help new contributors the same ways EddieHub helped me 🤝 I'll be more than happy to assist EddieHub in organising this mega-event.

Looking for suggestions/other angles to this, Thanks!

**Comment by suitupstan** [comment]
Hey 👋 Just looking forward to seeing how many people would be interested in being a part of this event, react/comment with "👋", to show your interest

**Reply by ewainy** [reply]
👋 Super interested in being part of this event and doing the challenge, sounds fun and lots to learn 😊💃

**Reply by eddiejaoude** [reply]
Awesome! I will ad some notes and thoughts soon 🤓

**Reply by kiritocode1** [reply]
im really really excited to work under you man , lets goo

**Reply by meenal21** [reply]
👋 Excited to get started on this!!

**Reply by CBID2** [reply]
👋🏾 I'm totally up for #100daysofopen source @krshkun !

**Comment by eddiejaoude** [comment, marked as answer]
Great write up @krshdev 👍  Here are my thoughts...

### 7 day open source contributor challenge

This is achievable and also encourages little and often with rest, instead of burnout

- day 1: find a project and read the README.md
- day 2: add value to someone's issue by commenting on it
- day 3: create an issue
- day 4: make changes
- day 5: create a pull request for changes
- day 6: review someone else's pull request (must leave a comment)
- day 7: rest  

Then repeat and help someone else to do the same.

### 7 day open source maintainer challenge

- day 1: create a repo with description and topics, turn off sections not being used (wiki, packages ...)
- day 2: add a basic README.md (goal of project, description ... )
- day 3: add issues with labels of what needs to be build (with project board)
- day 4: build the foundations of the project: hello world, templates
- day 5: add more to the foundations with testing, github actions etc
- day 6: improve the readme with new features
- day 7: rest

**Reply by suitupstan** [reply]
Hey @eddiejaoude, what I mean is, instead of having 7 days of non-stop challenges, we have 1 set of challenges per week, and a total of 14 challenges, it is exactly this but with more flexibility and rest! Plus 7 days doesn't make it sort of a challenge, it is more like a hackathon^^

What I thought of when I came to this idea was:

# First Month: Exploration Rocks!

- The first week, create a github account, create a github profile readme and join EddieHub
- The second week, explore github issues and add value to a project by creating an issue
- The third week, explore pull requests, review pull requests, create pull requests, etc!
- The fourth week, project boards, connecting with other participants, and basically, have fun for the completion of month 1!

# Second Month: Contributing to OpenSource 👀 

- The first week, Make a contribution to a project you use 
- The second week, Participants can share their github profiles, and you add value to 2 repos from different participants
- The third week, Exploring GitHub Actions & Pages. Deploying a site with pages & contributing an action to a repository
- The fourth week, connect with other participants, hang out, have office hours, and have fun for the completion of month 2!

# Third Month: Building & Maintaining Open Source Projects

- The first week, connect, ideate, innovate, build teams
- The second week, build the project, write readmes, issue templates, documentation etc
- The third week, improve developer experience, through github, actions, packages, etc
- The fourth week, connect with participants, have office hours, share your project, make contributions to others' projects. and have fun for the completion of the challenges!

# The rest 10 days.

- The participants can rest now, share their experiences via blogs, tweets etc (BONUS POINTS 🥳)
- The organising team (because this might be a huge event with hundreds of participants we may need one 🙃) will evaluate the submissions, judge the projects (based on how well open-sourced it was!)
- Participant-Organised Meetups/Calls
- Closing ceremony, with Eddie 😱 🤩 


> This way the people get tons of rest because they just need to devote about 1-2 hours per week, instead of per day, and more flexibility because it can be done at any day of the week as per their will.
> Plus, this way we create more sense of collaboration and learning than the competition or rush

Also @eddiejaoude it is @krshdev, you mentioned some one else 😅

**Comment by mikeysan** [comment]
I updated Eddie's comment to fix the mismatched name tagging

**Reply by eddiejaoude** [reply]
Thanks Mikey!

**Comment by CBID2** [comment]
I like this idea @krshkun! I noticed that people create repos with templates when doing the #100days of code challenge. I think there should be a similar approach for the #100daysofopensource. Let me know what you think.

**Reply by AbdulMuhaimin-1** [reply]
Hello @CBID2,  please i really wish to join in this event, i just saw this but wondering if it is a passed or an upcoming event...

**Reply by CBID2** [reply]
I don't think it was done

**Reply by AbdulMuhaimin-1** [reply]
So i guess it's yet to happen right, because i really wish i could go through this to learn more about open source... I'm a beginner to contributing to opensource..
